power-pay-frontend/src/Hooks/StorageContext.tsx

```diff
@@ -0,0 +1,236 @@
+import {createContext, PropsWithChildren, useEffect, useState} from 'react';
+
+
+// Define the StorageContextData interface.
+export interface StorageContextData<T> {
+    item: Record<string, T>;
+    setItem: (key: string, value: T) => Promise<boolean>;
+    removeItem: (key: string) => Promise<boolean>;
+    clear: () => Promise<boolean>;
+}
+
+// Define the StorageService interface.
+interface StorageService {
+    clear: () => Promise<boolean>;
+    getItem: <T>(key: string) => Promise<T | undefined>;
+    removeItem: (key: string) => Promise<boolean | void>
+
+
+
+}
+interface SetStorage {
+    setItem: <T>(key: string, value: T) => Promise<boolean | void>;
+}
+interface SetStorageIDB {
+    setItem: <T>(value: object, key: IDBValidKey) => Promise<boolean | void>;
+}
+
+// Define the LocalStorageService class implementing the StorageService interface.
+export class LocalStorageService implements StorageService, SetStorage {
+    async getItem<T>(key: string): Promise<T | undefined> {
+        const found = localStorage.getItem(key);
+        if (!found) return undefined;
+        return JSON.parse(found) as T;
+    }
+
+    async setItem<T>(key: string  , value: T): Promise<boolean | void> {
+        localStorage.setItem(key, JSON.stringify(value));
+        return true;
+    }
+
+    async removeItem(key: string): Promise<boolean> {
+        localStorage.removeItem(key);
+        return true;
+    }
+
+    async clear(): Promise<boolean> {
+        localStorage.clear();
+        return true;
+    }
+}
+
+export class IndexedDBStorageService implements StorageService, SetStorageIDB {
```

Comment:
hello @Christiantyemele, you implemented the  service IndexedDBStorageService, but the branch on the usestorage hook hasnt been merged yet, so if you merge this, it will arise to a merge conflict,
so go through the usestorage branch and ensure everything there works first before you can add the IndexedDBStorageService for the indexeddb

Reply:
here is the PR @Christiantyemele  https://github.com/ADORSYS-GIS/e2e-banking-app/pull/83